### ISSUE TITLE
Documentation: source-layer (third party vector-tile sources, v0.11.4)

### ISSUE THREAD

**Issue by Dirk--** [mapbox/mapbox-gl-js#1734]
"source-layer" seems to be mandatory:
works:
https://vector.mapzen.com/osm/all/12/2096/1365.mvt
( https://vector.mapzen.com/osm/all/12/2096/1365.json for refeference)
fails:
https://vector.mapzen.com/osm/water/12/2096/1365.mvt
( https://vector.mapzen.com/osm/water/12/2096/1365.json for refeference)

**Comment by tmcw** [comment]
Can you add a bit more detail here? `source-layer` is a style property, and you've provided examples of data. For any _vector_ source, source-layer is required, for raster sources it isn't, since they don't contain source-layers.

**Comment by Dirk--** [comment]
I would like to point out, that mapbox-gl-js seems to need a .mvt having a 'layer' information. 

working:
```
    "sources": {
        "mapzen-all": {
            "type": "vector",
            "tiles": ["https://vector.mapzen.com/osm/all/{z}/{x}/{y}.mvt"]
        }
    }
     "layers": [{
            "id": "water",
            "type": "fill",
            "source": "mapzen-all",
            "source-layer": "water"
    }]
```
fails:
```
    "sources": {
        "mapzen-water": {
            "type": "vector",
            "tiles": ["https://vector.mapzen.com/osm/water/{z}/{x}/{y}.mvt"]
        }
    }
    "layers": [{
            "id": "water",
            "type": "fill",
            "source": "mapzen-water"
    }]
```

**Comment by bhousel** [comment]
Maybe relates to #1576, cc @zerebubuth, who knows more about the Mapzen vector source

**Comment by lucaswoj** [comment]
You are correct -- all layers that use a vector source must specify a `source-layer`, even if the vector source only has one layer. The "failing" example above should work as 

```
    "sources": {
        "mapzen-water": {
            "type": "vector",
            "tiles": ["https://vector.mapzen.com/osm/water/{z}/{x}/{y}.mvt"]
        }
    }
    "layers": [{
            "id": "water",
            "type": "fill",
            "source": "mapzen-water",
            "source-layer": "water"
    }]
```

ref #1637

## Next Steps

 - [ ] Display an error message if no `source-layer` is specified for a layer that uses a vector source.
 - [ ] Ensure docs clearly state that `source-layer` is required when using a vector source

**Comment by lucaswoj** [comment]
This was resolved by mapbox/mapbox-gl-style-spec#381
